addons/primitives/high_primitives.c

```diff
@@ -455,9 +449,14 @@ void al_calculate_arc(float* dest, int stride, float cx, float cy,
          for (ii = 0; ii < num_points; ii ++) {
             *dest =       r2 * x + cx;
             *(dest + 1) = r2 * y + cy;
+            *(dest + 3) = r2 * x;
```

Comment:
Unfortunately this isn't backwards compatible, we can't change what this function is doing, since existing users could have been just passing 2*num_points floats and no more.

This will need to be computed after the fact in al_draw_circle etc (by subtracting away cx/cy from x/y)